src/settings/JobProfiles/NewJobProfileRoute/NewJobProfileRoute.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { isEmpty } from 'lodash';
+
+import { stripesConnect } from '@folio/stripes/core';
+
+import { JobProfilesForm } from '../JobProfilesForm';
+
+const getFormattedMappingProfiles = mappingProfiles => {
+  if (isEmpty(mappingProfiles?.records)) return [];
+
+  return mappingProfiles.records.reduce((memo, current) => {
+    memo.push({
+      label: current.name,
+      value: current.id,
+    });
+
+    return memo;
+  }, []);
+};
+
+const NewJobProfileRoute = props => {
+  const {
+    onCancel,
+    onSubmit,
+    resources: { mappingProfiles },
+  } = props;
+
+  return (
+    <JobProfilesForm
+      onSubmit={onSubmit}
+      hasLoaded={mappingProfiles && mappingProfiles.hasLoaded}
+      mappingProfiles={getFormattedMappingProfiles(mappingProfiles)}
```

Comment:
```suggestion
      mappingProfiles={getFormattedMappingProfiles(mappingProfiles?.records)}
```

So we can simplify (omit - if (isEmpty(mappingProfiles?.records)) return [];) by providing default parameter.
```
      const getFormattedMappingProfiles = (mappingProfiles = []) => {
```